src/ansiblelint/utils.py

```diff
@@ -759,6 +761,60 @@ def construct_mapping(
     return data
 
 
+def iter_tasks_in_file(
+    lintable: Lintable, rule_id: str
+) -> Iterator[Tuple[Dict[str, Any], Dict[str, Any], bool, Optional[MatchError]]]:
+    """Iterate over tasks in file.
+
+    This yields a 4-tuple of raw_task, normalized_task, skipped, and error.
+
+    raw_task:
+        When looping through the tasks in the file, each "raw_task" is minimally
+        processed to include these special keys: __line__, __file__, skipped_rules.
+    normalized_task:
+        When each raw_task is "normalized", action shorthand (strings) get parsed
+        by ansible into python objects and the action key gets normalized. If the task
+        should be skipped (skipped is True) or normalizing it fails (error is not None)
+        then this is just the raw_task instead of a normalized copy.
+    skipped:
+        Whether or not the task should be skipped according to tags or skipped_rules.
+    error:
+        This is normally None. It will be a MatchError when the raw_task cannot be
+        normalized due to an AnsibleParserError.
+
+    :param lintable: The playbook or tasks/handlers yaml file to get tasks from
+    :param rule_id: The current rule id to allow calculating skipped.
+
+    :return: raw_task, normalized_task, skipped, error
+    """
+    data = parse_yaml_linenumbers(lintable)
+    if not data:
+        return
```

Comment:
codecov doesn't like this. I could increase coverage by indenting the rest of this function under `if data:` ... but that's kind  of gross. Can we take the hit in coverage? Or do you have a recommended way to add a test for this?

Reply:
Considering the nature of the change it would be wise to add a test for it. I guess that adding some `foo: none` in a playbook would do the trick. You may be able to do this even if you modify an existing test, i guess.

Reply:
ok. I added a test.